packages/lib/passport-interface/src/PassportPopup/react.ts

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from "react";
+import { receiveZupassPopupMessage, zupassPopupSetup } from "./core";
+
+/**
+ * React hook that listens for PCDs and PendingPCDs from a Zupass popup window
+ * using message passing and event listeners.
+ */
+export function useZupassPopupMessages(): [string, string] {
+  const [pcdStr, setPCDStr] = useState("");
+  const [pendingPCDStr, setPendingPCDStr] = useState("");
+
+  // Listen for PCDs coming back from the Zupass popup
+  useEffect(() => {
+    const abortReceiveMessage = new AbortController();
+    receiveZupassPopupMessage(abortReceiveMessage.signal).then((result) => {
+      if (result.type === "pcd") {
+        setPCDStr(result.pcdStr);
+      } else if (result.type === "pendingPcd") {
+        setPendingPCDStr(result.pendingPcdStr);
+      }
+    });
+    return () => {
+      // If the hook is unmounted, signal that the message handlers can detach
+      // by aborting.
+      abortReceiveMessage.abort();
+    };
+  }, []);
+
+  return [pcdStr, pendingPCDStr];
+}
+
+/**
+ * A react hook that sets up necessary Zupass popup logic on a specific route.
```

Comment:
`necessary` - for what?

Reply:
I've simplified the comment here, and referenced `zupassPopupSetup`, so that there's one place to look for information about the /popup route.